openedx/core/djangoapps/user_api/views.py

```diff
@@ -206,10 +207,21 @@ def __init__(self, *args, **kwargs):
 
         # Map field names to the instance method used to add the field to the form
         self.field_handlers = {}
-        for field_name in self.DEFAULT_FIELDS + self.EXTRA_FIELDS:
+        valid_fields = self.DEFAULT_FIELDS + self.EXTRA_FIELDS
+        for field_name in valid_fields:
             handler = getattr(self, "_add_{field_name}_field".format(field_name=field_name))
             self.field_handlers[field_name] = handler
 
+        field_order = configuration_helpers.get_value('REGISTRATION_FIELD_ORDER')
+        if not field_order:
+            field_order = settings.REGISTRATION_FIELD_ORDER or valid_fields
+
+        # Check that all of the valid_fields are in the field order and vice versa, if not set to the default order
+        if set(valid_fields) != set(field_order):
```

Comment:
I decided to make life simpler and just check that they each have the same set of fields - and if not, go to the default order. Maybe that's overly simple but it's in the same spirit as what I was doing before.